src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorRuntimeNodeWriter.cs

```diff
@@ -586,64 +598,89 @@ private static string GetContent(HtmlContentIntermediateNode node)
             return builder.ToString();
         }
 
-        private static void WriteAttributeValue(CodeWriter writer, object value)
+        // There are a few cases here, we need to handle:
+        // - Pure HTML
+        // - Pure CSharp
+        // - Mixed HTML and CSharp
+        //
+        // Only the mixed case is complicated, we want to turn it into code that will concatenate
+        // the values into a string at runtime.
+
+        private static void WriteAttributeValue(CodeWriter writer, IList<IntermediateToken> tokens)
```

Comment:
I hadn't realized that you had support for these cases 😆 

I had to revisit this code because it wasn't handling a few cases correctly, most notably when you have consecutive CSharp tokens, which happens now for bind and event handlers.